Refactored client. Updated client model. Added unit test for client.

FiwareULPGC opened the pull request:
This PR implements the following:
- Updated client model in order to use the last API changes.
- Added client unit tests.
- Added restaurant rate in the restaurant popup.
- Added new menu entry with new operations: My organizations.
- Fixed ShowMessage behaviour when multiple messages are raised.
- Client refactored in order to separate the logic, the API calls and the rendering.
- Divided utils file in utils and AJAXRequest.
- Added comments to modules.
- Cleaned code and fixed format.
- Updated jquery timepicker.
- Added minor input checks for reservations.

FiwareULPGC committed:
Updated client model in order to use the last API changes.
Added client unit tests.
Added restaurant rate in the restaurant popup.
Added new menu entry with new operations: My organizations.
Fixed ShowMessage behaviour when multiple messages are raised.
Client refactored in order to separate the logic, the API calls and the rendering.
Divided utils file in utils and AJAXRequest.
Added comments to modules.
Cleaned code and fixed format.
Updated jquery timepicker.
Added minor input checks for reservations.

jmcanterafonseca commented on client/all_restaurants_and_reviews.html:
```diff
@@ -1,70 +0,0 @@
-    <link href="stylesheets/styles.css" rel="stylesheet">
-  
-    <!-- Addons Styles -->
-    <!--<link href="addons/.../css/file.css" rel="stylesheet"> -->
```
code commented ...

jmcanterafonseca commented on client/all_restaurants_and_reviews.html:
```diff
@@ -1,70 +0,0 @@
-   </div> 
-</div>
-</body>
-</html>
```
nit: new line

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+      function(response) { //error
+        alert('Could not retrieve restaurants');
+        if (response) {
+          console.log(response);
```
console.error 

and you would need to log something more meaningful

console.error('Error while xxx : ' , response.status, response.<whatever>), something that allows to debug what is happening

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+      },
+      function(error) {
+        alert('Cannot get user reviews for: ' + username);
+        console.log(error);
```
console.error and more info as suggested above

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+      },
+      function(error) {
+        alert('Cannot get user reviews for: ' + organization);
+        console.log(error);
```
console.error

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+  }
+
+  function showReview(reviewId) {
+
```
nit: no newline needed, same for the rest of functions

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+      alert('Cannot get the specified review');
+      console.log(err);
+    });
+
```
no new line needed

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+        drawModule.setPopupTitle(
+          'Edit Review for ' + review.itemReviewed.name);
+        drawModule.setPopupContent(formDiv);
+        drawModule.inicializeReviewForm(review);
```
function name is incorrect `initialize`should be

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+  }
+
+  function createNewReservation(name, partySize, time) {
+
```
unneeded empty line

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,280 @@
+      drawModule.closePopUpWindow,
+      function(err) {
+        alert('Cannot add reservation');
+        console.log(err);
```
console.error

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -19,8 +21,9 @@ var connectionsAPI = (function() {
     endUser: 'End user',
-    restaurantViewer: 'Restaurant Viewer',
-    globalManager: 'global manager'
+    restaurantViewer: 'Restaurant viewer',
```
end user?

FiwareULPGC replied:
It is specified in the FIWARE Tour Guide App - Functional Specification.
https://docs.google.com/document/d/1FpeuBzPkAkIs90rP16jeOdi5-JuSbcdsiOatE87l3Dw/edit

 It is also returned from the implemented API. For example user1 has the roles:

```
"roles":[{"name":"End user","id":"f715a71095c447e7bd5bccb1e65d4bf5"}]
```

jmcanterafonseca replied:
ok

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -171,9 +214,11 @@ var connectionsAPI = (function() {
-    for (var index = 0, len = userInfo.roles.length; index < len; ++index) {
-      if (role == userInfo.roles[index].name) {
-        return true;
+    if (userInfo) {
```
use Array.filter simpler and easier to read

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -87,56 +88,98 @@ var connectionsAPI = (function() {
+          organizationSubMenu.className = 'dropdown-menu';
+
+          //restaurants submenu
+          var restaurantLi = document.createElement('LI');
```
this code is duplicated three times, here and two additional times below

perhaps you can create a function for it

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -203,13 +248,13 @@ var connectionsAPI = (function() {
 
   function loginNeeded(action) {
-    if (null != localStorage.getItem('userInfo')) {
+    if (null != getUser()) {
```
better the other way round, if (getUser() != null)

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -219,12 +264,31 @@ var connectionsAPI = (function() {
+
+  function is_organization_manager(organization) {
+    if (organization.roles) {
+      for (var i = organization.roles.length - 1; i >= 0; i--) {
```
Array.filter

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+var utils;
+var drawModule = (function() {
+
+  var maxRating = 5;
```
if this is a constant, then user upper case

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+  };
+
+  //actions do nothing by default
+  var viewReservationAction = function() {};
```
instead of repeating this son many times, better

var NOP = function() {};

and then var x = NOP, y= etc

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+  /*this functions add the restaurants to the leaflet map */
+  function addRestaurantstoMap(restaurants) {
+
```
unneeded new line

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    popHTML.appendChild(showReservations);
+    popHTML.appendChild(document.createElement('BR'));
+
+
```
too many blank lines

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+  /*show restaurant reviews from an API response */
+  /* At this moment, show all reviews without pagination */
+  function createReviewsDiv(reviewsResponse) {
+    reviewsResponse = JSON.parse(reviewsResponse);
```
too many blank lines

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+      var authorValue = document.createElement('SPAN');
+      authorValue.className = 'authorValue';
+      authorValue.textContent = reviewsResponse[j].author.name;
```
consider reusing code through a function

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    timeHead.textContent = 'Reservation time:';
+    row.appendChild(timeHead);
+
+    var dinersHead = document.createElement('TH');
```
nit: dinners

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    reservationsTable.appendChild(tableBody);
+    return reservationsTable;
+  }
+
```
unneeded new line

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    return reviewForm;
+  }
+
+  //inicialize a review form when updating a review
```
nit: initialize

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+  //inicialize a review form when updating a review
+  //instead of create a new one
+  function inicializeReviewForm(review) {
```
function name is wrong, should be initialize

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+    var createReservationLink = document.createElement('A');
+    createReservationLink.textContent = 'Make a reservation';
+    createReservationLink.onclick = (function(restaurantName) {
```
better to use bind instead of the closure

``` js
createReservation.onclick = function(restaurantName)  {
}.bind(null, restaturantName)
```

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+
+  function createReservationForm(restaurantName) {
+
```
unneeded new line

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+      createAndShowReviewForm(restaurantName);
+      };
+    })(restaurantName);
+
```
better with bind

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    name.value = restaurantName;
+    reservationForm.appendChild(name);
+
+    var dinersLabel = document.createElement('SPAN');
```
dinnersLabel

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+  }
+
+
+  function enableAvailabilityCalculation() {
```
unneeded line

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+      }
+    }
+
+    var nDiners = document.getElementById('partySize').valueAsNumber;
```
numDinners

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+    //if available allow to submit
+    document.getElementById('submitReservation').disabled =
+      ! available;
```
better !available

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    document.getElementById('submitReservation').disabled =
+      ! available;
+
+      if (! available) {
```
unneeded space

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+          (capacity - occupancyLevel) +
+          ' seats available for the specified date and time';
+        utils.showMessage(errMsg, 'alert-warning');
+      }
```
unneeded space

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+      'has-error');
+    document.getElementById('reservationTime').classList.remove('has-error');
+    return true;
+
```
unneeded space

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    document.getElementById('reviewsTableBody').appendChild(row);
+  }
+
+  //create a table for displaying the organization reviews.
```
for multiline comments use 
/*
 */

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+    var reviewList = document.createElement('DIV');
+    reviewList.className = 'reviewList';
+
+    for (var j = 0, lim = reviewsResponse.length; j < lim; j++) {
```
better Array.forEach

jmcanterafonseca commented on client/js/drawModule.js:
```diff
@@ -0,0 +1,1073 @@
+
+    var tableBody = document.createElement('TBODY');
+
+    for (var j = 0, lim = reservationsResponse.length; j < lim; j++) {
```
Array.forEach

jmcanterafonseca commented on client/js/organizationReservations.js:
```diff
@@ -0,0 +1,41 @@
+  clientLogic.setUpDrawModule();
+
+  //get franchise from url
+  var franchise = window.location.search.replace('?', '');
```
I think we need to find another way to detect what is the current franchise rather than rely on the URL

FiwareULPGC replied:
We can try to pass the franchise via sessionStorage. So we should change the current menu link to a function that stores the target franchise and then perform a redirection using, for example, window.location.

Do you agree with this approach?

jmcanterafonseca replied:
yes, thanks

jmcanterafonseca commented on client/js/organizationReviews.js:
```diff
@@ -0,0 +1,37 @@
+    });
+
+  }
+
```
same comment and above and code is duplicated here

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -8,97 +8,89 @@
   }
 
-  function getOrganizationRestaurants(organization) {
+  function getOrganizationRestaurants(organization, cb, err_cb) {
```
nit: `err_cb` should be `errorCb`

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -8,97 +8,89 @@
+    restaurants.forEach(function(restaurant) {
+      convertedRestaurants.push(
+        convertRestaurant(restaurant)
+        );
```
probably here you need to use Array.map

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -174,310 +168,28 @@ var restaurantsAPI = (function() {
-    popHTML.appendChild(showReviews);
-    popHTML.appendChild(document.createElement('BR'));
+  function getRestaurantReservationsByDate(restaurantName, time,
+    cb, err_cb) {
```
errorCb

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -174,310 +168,28 @@ var restaurantsAPI = (function() {
-    var ratingValue = parseInt(document.forms.editReviewForm.ratingValue.value);
-    var reviewBody = document.forms.editReviewForm.reviewBody.value;
+  function createNewReview(
+    restaurantName, ratingValue, reviewBody, cb, err_cb) {
```
nit: it fits in the same line

jmcanterafonseca commented on client/myReviews.html:
```diff
@@ -23,17 +23,19 @@
   <script src="js/connectionsAPI.js"></script>
   <script src="js/myReviews.js"></script>
+  <script src="js/clientLogic.js"></script>
+  <script src="js/drawModule.js"></script>
```
if these scripts do not have any DOM dependency then it is better to load them with <script defer>

jmcanterafonseca commented on client/organizationReservations.html:
```diff
@@ -0,0 +1,87 @@
+  <script src="js/organizationReservations.js"></script>
+  <script src="js/clientLogic.js"></script>
+  <script src="js/drawModule.js"></script>
+
```
for consistency here you would need to add the type attribute as well

jmcanterafonseca commented on client/index.html:
```diff
@@ -37,9 +37,12 @@
   <script src="js/index.js"></script>
   <script src="js/connectionsAPI.js"></script>
+  <script src="js/clientLogic.js"></script>
+  <script src="js/drawModule.js"></script>
```
script type missing

consider use 

`<script defer>`

jmcanterafonseca commented on client/stylesheets/app.css:
```diff
@@ -254,8 +259,64 @@
+    margin-right: -10px;
+}
+
+.dropdown-submenu:hover>a:after {
```
nit: use space  `:hover > a:after`

jmcanterafonseca commented on client/stylesheets/app.css:
```diff
@@ -254,8 +259,64 @@
+    display: block;
+}
+
+.dropdown-submenu>a:after {
```
nit: spaces

jmcanterafonseca commented on client/spec/test_utils.js:
```diff
@@ -0,0 +1,73 @@
+      var navBar = document.getElementById('topMenu');
+      var messageDiv = navBar.nextSibling;
+
+      expect(messageDiv).to.have.property('className').that.is.a('string').to.contain('alert').to.contain('fade').to.contain('in').to.contain('alert-warning');
```
line too long

jmcanterafonseca commented on client/spec/test_utils.js:
```diff
@@ -0,0 +1,73 @@
+      var navBar = document.getElementById('topMenu');
+      var messageDiv = navBar.nextSibling;
+
+      expect(messageDiv).to.have.property('className').that.is.a('string').to.contain('alert').to.contain('fade').to.contain('in').to.contain('alert-warning');
```
line to long

jmcanterafonseca commented on client/spec/test_utils.js:
```diff
@@ -0,0 +1,73 @@
+  describe('Testing showMessage function', function() {
+
+    beforeEach(function() {
+
```
unneeded line

jmcanterafonseca commented on client/spec/test_utils.js:
```diff
@@ -0,0 +1,73 @@
+
+    beforeEach(function() {
+
+      var doc = jsdom("<div class='container-fluid'>"+
```
nit space after `"`and before `+`

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/organization/'+organizationName);
+        this.requests[0].method.should.equal( 'GET');
+        this.requests[0].respond(200, { 'Content-Type': 'text/json' }, dataJson);
```
text/json is not a proper MIME type. it should be `application/json`

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/user/'+username);
+        this.requests[0].method.should.equal( 'GET');
+        this.requests[0].respond(200, { 'Content-Type': 'text/json' }, dataJson);
```
`text/json` does not exist

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+
+
+    it('Simplify restaurants format', function() {
+        
```
nit: unneeded line

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+
+
+    it('Create a reservation', function() {
+      
```
unneeded line

jmcanterafonseca commented on client/spec/test_utils.js:
```diff
@@ -0,0 +1,73 @@
+      
+      expect(messageDiv).to.have.property('className').that.is.a('string').to.contain('alert').to.contain('fade').to.contain('in').to.contain('alert-danger');
+      expect(messageDiv).to.have.property('textContent').that.is.a('string').to.be.equal(message+'X')
+    
```
nit: unneeded lines

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+          done();
+        });
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/organization/'+organizationName);
```
instead of repiting http://tourguide/api/... etc., why don't you create a constant, at least for the base URL?

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+      restaurantsAPI.createNewReservation(restaurantName,partySize,reservationDatetime, function(){
+          assert(true, 'Success function called');
+        }, 
+        function(){
```
nit: space between function() and curly brace

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+        expect(result).to.be.deep.equal(expectedResult);
+
+    });
+
```
nit: too many spaces

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+
+
+var restaurantsJSON = [{
+      "@context": "http://schema.org",
```
I would suggest this JSON data is moved to a different file

jmcanterafonseca commented on client/spec/test_restaurantsAPI.js:
```diff
@@ -0,0 +1,534 @@
+      "name": "user1"
+    }
+  }
+];
```
consider moving to a different file a data only file

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+      drawModule.inicializeReviewForm(singleReviewJSON);
+      expect(document.getElementsByName('reviewBody')[0].value).to.be.equal(singleReviewJSON.reviewBody);
+      expect(document.getElementsByName('ratingValue')[0].value).to.be.equal(singleReviewJSON.reviewRating.ratingValue+'');
```
nit: unneeded spaces

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+        return document.forms[0];
+      }
+
+      drawModule.inicializeReviewForm(singleReviewJSON);
```
initialize

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      expectedForm.appendChild(textArea);
+
+      expectedForm.appendChild(document.createElement('BR'));
+
```
maybe you can create the expected form using jsdom as well

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      var ratLab = document.createElement('LABEL');
+      ratLab.textContent = 'Rating value: ';
+      expectedForm.appendChild(ratLab);
+
```
consider using jsdom for the expected form

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+    expectedEle.appendChild(tr2);
+
+
```
unneeded spaces

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+    tr2.appendChild(viewTd2);
+    tr2.appendChild(editTd2);
+    tr2.appendChild(deleteTd2);
+
```
consider using jsdom instead of creating manually the DOM elements

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      drawModule.createReservationsTable(reservationsResponse);
+
+     expect(document.getElementById('reservationsTableBody').innerHTML).to.be.deep.equal(expectedEle.innerHTML);
+
```
nit: unneeded empty line

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      tr1.appendChild(res1);
+      tr1.appendChild(date1);
+      tr1.appendChild(diners1);
+      tr1.appendChild(cancel1);
```
consider using jsdom

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+      expectedDiv.appendChild(thead);
+      expectedDiv.appendChild(tbody);
+
```
consider using jsdom

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+
+    it('Test create reservations div', function() {
+      
```
unneeded line

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      review2.appendChild(body2);
+
+      expectedDiv.appendChild(review2);
+
```
consider using jsdom

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      drawModule.setPopupContent(nDiv);
+
+      expect(document.getElementById('popContent').children[0]).to.be.deep.equal(nDiv);
+
```
unneeded line

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+      drawModule.setPopupTitle(title);
+
+      expect(document.getElementById('popTitle').textContent).to.be.equal(title);
```
unneeded line

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+var time2 = "2015-11-06T21:57:22.604Z";
+
+var reviewsJSON = [
+  {
```
use different file for mock data

jmcanterafonseca commented on client/spec/test_AJAXRequest.js:
```diff
@@ -0,0 +1,94 @@
+    });
+
+
+})
```
new line needed

jmcanterafonseca commented on client/organizationReservations.html:
```diff
@@ -0,0 +1,87 @@
+  </div>
+</div>
+</body>
+</html>
```
new line needed

jmcanterafonseca commented on client/organizationRestaurants.html:
```diff
@@ -36,9 +35,12 @@
+  <script src="js/organizationRestaurants.js"></script>
   <script src="js/connectionsAPI.js"></script>
+  <script src="js/clientLogic.js"></script>
+  <script src="js/drawModule.js"></script>
```
script defer

jmcanterafonseca commented on client/organizationReviews.html:
```diff
@@ -0,0 +1,83 @@
+  <script src="js/connectionsAPI.js"></script>
+  <script src="js/organizationReviews.js"></script>
+  <script src="js/clientLogic.js"></script>
+  <script src="js/drawModule.js"></script>
```
type needed. script defer should be considered

jmcanterafonseca commented on client/organizationReviews.html:
```diff
@@ -0,0 +1,83 @@
+  </div>
+</div>
+</body>
+</html>
```
new line needed

jmcanterafonseca commented on client/spec/helpers/chai.js:
```diff
@@ -0,0 +1,6 @@
+global.sinon = require('sinon');
+global.expect = chai.expect;
+global.AssertionError = chai.AssertionError;
+global.Assertion = chai.Assertion;
```
new line needed

jmcanterafonseca commented on client/spec/helpers/jsdom.js:
```diff
@@ -0,0 +1,8 @@
+} else {
+  window.expect = window.chai.expect
+  window.require = function () { /* noop */ }
+}
```
new line needed

jmcanterafonseca commented on client/spec/test_AJAXRequest.js:
```diff
@@ -0,0 +1,94 @@
+
+  beforeEach(function() {
+    this.xhr = sinon.useFakeXMLHttpRequest();
+
```
nit: unneeded line

jmcanterafonseca commented on client/spec/test_AJAXRequest.js:
```diff
@@ -0,0 +1,94 @@
+  afterEach(function() {
+    this.xhr.restore();
+  });
+
```
no new line  needed

jmcanterafonseca commented on client/spec/test_AJAXRequest.js:
```diff
@@ -0,0 +1,94 @@
+        done();
+      });
+
+      this.requests[0].respond(200, { 'Content-Type': 'text/json' }, dataJson);
```
application/json

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+      global.document = window.document;
+      global.navigator = {
+            userAgent: 'nodejs'
+          }
```
nit: indentation, use something like

``` js
global.navigator = {
   userAgent: 'node.js'
}
```

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+
+      document.getElementsByClassName('container-fluid')[0].appendChild(testForm);
+
+      //hacked because it fails using JSDOM
```
space between // and comment

jmcanterafonseca commented on client/spec/test_drawModule.js:
```diff
@@ -0,0 +1,745 @@
+  })
+
+
+})
```
new line needed

jmcanterafonseca commented:
please check

* unneeded new lines
* indentation
* code reuse
* mock data
* Use Array.forEach, Array.map, Array.filter
* Use of script defer and script type
* new line at the end of file

I have spotted all the major places where it is wrong but there can be others. 

Overall looks good, let's have a new round, 

thanks!

FiwareULPGC committed:
Usage of Array.forEach, Array.map, Array.filter.
Corrections to code style.
Code reuse.
Mock data moved to separate files.
Usage of script defer and script type.
Franchise param passed using sessionStorage.

jmcanterafonseca commented on client/js/clientLogic.js:
```diff
@@ -0,0 +1,296 @@
+      console.log('Error retrieving the review: ' + reviewId);
+      console.log(err);
+    });
+
```
new line not needed

jmcanterafonseca replied:
I'm still seeing console.log where console.error , please could you check that you address all comments? 

thanks

FiwareULPGC committed:
console.log() replaced by console.error().
alert() replaced by utils.showMessage().
Fixed blank lines.

jmcanterafonseca commented:
LGTM
